core/templates/domain/question/question-update.service.ts

```diff
@@ -16,165 +16,191 @@
  * @fileoverview Service to handle the updating of a question.
  */
 
-import { Change } from
+import { BackendChangeObject, Change, DomainObject } from
   'domain/editor/undo_redo/change.model';
 import { Interaction } from 'domain/exploration/InteractionObjectFactory';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import {UndoRedoService} from 'domain/editor/undo_redo/undo-redo.service';
+import {QuestionDomainConstants} from 'domain/question/question-domain.constants';
+import {Question} from 'domain/question/QuestionObjectFactory';
 
-require('domain/editor/undo_redo/question-undo-redo.service.ts');
-require('domain/editor/undo_redo/undo-redo.service.ts');
-require('domain/question/QuestionObjectFactory.ts');
+@Injectable({
+  providedIn: 'root'
+})
+export class QuestionUpdateService {
+  private CMD_UPDATE_QUESTION_PROPERTY: string;
+  private QUESTION_PROPERTY_INAPPLICABLE_SKILL_MISCONCEPTION_IDS: string;
+  private QUESTION_PROPERTY_LANGUAGE_CODE: string;
+  private QUESTION_PROPERTY_QUESTION_STATE_DATA: string;
 
-require('domain/question/question-domain.constants.ajs.ts');
+  constructor(private QuestionUndoRedoService: UndoRedoService) {
+    this.CMD_UPDATE_QUESTION_PROPERTY = (
+      QuestionDomainConstants.CMD_UPDATE_QUESTION_PROPERTY);
+    this.QUESTION_PROPERTY_INAPPLICABLE_SKILL_MISCONCEPTION_IDS = (
+      QuestionDomainConstants.
+        QUESTION_PROPERTY_INAPPLICABLE_SKILL_MISCONCEPTION_IDS
+    );
+    this.QUESTION_PROPERTY_LANGUAGE_CODE = (
+      QuestionDomainConstants.QUESTION_PROPERTY_LANGUAGE_CODE
+    );
+    this.QUESTION_PROPERTY_QUESTION_STATE_DATA = (
+      QuestionDomainConstants.QUESTION_PROPERTY_QUESTION_STATE_DATA
+    );
+  }
 
-angular.module('oppia').factory('QuestionUpdateService', [
-  'QuestionUndoRedoService',
-  'CMD_UPDATE_QUESTION_PROPERTY',
-  'QUESTION_PROPERTY_INAPPLICABLE_SKILL_MISCONCEPTION_IDS',
-  'QUESTION_PROPERTY_LANGUAGE_CODE', 'QUESTION_PROPERTY_QUESTION_STATE_DATA',
-  function(
-      QuestionUndoRedoService,
-      CMD_UPDATE_QUESTION_PROPERTY,
-      QUESTION_PROPERTY_INAPPLICABLE_SKILL_MISCONCEPTION_IDS,
-      QUESTION_PROPERTY_LANGUAGE_CODE, QUESTION_PROPERTY_QUESTION_STATE_DATA) {
-    var _applyChange = function(question, command, params, apply, reverse) {
-      var changeDict = angular.copy(params);
-      changeDict.cmd = command;
-      var changeObj = new Change(changeDict, apply, reverse);
-      QuestionUndoRedoService.applyChange(changeObj, question);
-    };
+  private applyChange(
```

Comment:
Here and below, prefer prefixing `_` for private functions as it was before (see [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript).)